examples/with-next/app/mockProvider.tsx

```diff
@@ -0,0 +1,31 @@
+'use client'
+import { useEffect, useState } from 'react'
+
+export function MockProvider({
+  children,
+}: Readonly<{
+  children: React.ReactNode
+}>) {
+  const [mockingEnabled, enableMocking] = useState(false)
+
+  useEffect(() => {
+    async function enableApiMocking() {
+      /**
+       * @fixme Next puts this import to the top of
+       * this module and runs it during the build
+       * in Node.js. This makes "msw/browser" import to fail.
+       */
+      const { worker } = await import('../mocks/browser')
```

Comment:
Next.js puts this dynamic import from the browser runtime to the Node.js build by moving it to the top of the module.

Reply:
How about fixing like this?

```tsx
if (typeof window !== 'undefined') {
  const { worker } = await import('../mocks/browser')
  await worker.start()
}
```

Reply:
This didn't work for me. I had to do this instead:
```tsx
if (process.env.NEXT_RUNTIME !== "nodejs") {
    const { worker } = await import("./browser");
    await worker.start({
      onUnhandledRequest: "bypass",
    });
  }
  ```

Reply:
@brycefranzen, that may actually work around webpack resolving this import in Node.js as well. I still think that's a bug though.